seedsource_core/django/seedsource/management/commands/import_seedzones.py

```diff
@@ -1,51 +1,103 @@
+from django.core.management import BaseCommand, CommandError
 from django.db import transaction
 from seedsource_core.django.seedsource.models import SeedZone, ZoneSource
 
-from ..utils import ZoneConfig
+from ..zoneconfig import ZoneConfig
 
-SEEDZONES_LOCATION = getattr(settings, 'SEEDZONES_LOCATION', 'data/seedzones')
+SEEDZONES_LOCATION = getattr(settings, "SEEDZONES_LOCATION", "data/seedzones")
 
 
 class Command(BaseCommand):
-    help = 'Loads polygon data from seed zone shape files into the database'
+    help = "Loads polygon data from seed zone shapefiles into the database."
 
     def add_arguments(self, parser):
-        parser.add_argument('zone_name', nargs='?', type=str)
-
-    def handle(self, zone_name, *args, **options):
-        with ZoneConfig(zone_name) as config:
-            source = ZoneSource.objects.get_or_create(name=config.source)[0]
-            if source.seedzone_set.all().exists():
-                message = (
-                    'WARNING: This will replace {} seed zone records and remove associated transfer limits. '
-                    'Do you want to continue? [y/n]'.format(config.label)
-                )
-                if input(message).lower() not in {'y', 'yes'}:
-                    return
-
-            self.stdout.write('Loading seed zones...')
-
-            with transaction.atomic():
-                source.seedzone_set.all().delete()
-                for polygon, info in config.get_zones():
-                    uid_suffix = 0
-                    while True:
-                        zone_uid = info['name'].format(zone_id=info['zone_id'])
-
-                        if uid_suffix > 0:
-                            zone_uid += '_{}'.format(uid_suffix)
+        parser.add_argument(
+            "zone_name",
+            nargs="?",
+            type=str,
+            default="",
+            help="Zone set name to import (default: import all in seedzones directory",
+        )
+
+        parser.add_argument(
+            "--overwrite",
+            dest="overwrite",
+            action="store_true",
+            default=False,
+            help="If True, will automatically delete and replace any existing zones",
+        )
+
+        parser.add_argument(
+            "--clear",
+            dest="clear",
+            action="store_true",
+            default=False,
+            help="If True, will automatically delete all zone sets and zones before importing new ones",
+        )
+
+    def handle(self, zone_name, overwrite, clear, *args, **options):
+        if clear:
+            message = (
+                "WARNING: This will delete all zone sets and zones, including all transfer limits and other data."
+                "Do you want to continue? [y/n]"
+            )
+            if input(message).lower() not in {"y", "yes"}:
+                return
+
+            self.stdout.write(
+                "Deleting all zone sets and zones.  This might take a while..."
```

Comment:
It _shouldn't_.

Reply:
It took a little bit, if I recall correctly.  I think my surprise at that was the motivation for adding the note.